contracts/strategy/rysk/RyskStrategy.sol

```diff
@@ -0,0 +1,234 @@
+        override(IStrategy)
+        returns (bool)
+    {
+        return _getTotalShares() != 0;
+    }
+
+    /// @inheritdoc IStrategy
+    function investedAssets()
+        external
+        view
+        virtual
+        override(IStrategy)
+        returns (uint256)
+    {
+        return _sharesToUnderlying(_getTotalShares());
+    }
+
+    /// @inheritdoc IStrategy
+    function invest() external virtual override(IStrategy) onlyManager {
+        uint256 balance = underlying.balanceOf(address(this));
+
+        if (balance == 0) revert StrategyNoUnderlying();
+
+        emit StrategyInvested(balance);
+
+        ryskLqPool.deposit(balance);
+    }
+
+    /// @inheritdoc IStrategy
+    function withdrawToVault(uint256 _amount)
+        external
+        virtual
+        override(IStrategy)
+        onlyManager
+    {
+        if (_amount == 0) revert StrategyAmountZero();
+
+        uint256 shares = _underlyingToShares(_amount);
+
+        _resetPendingWithdrawalIfEpochAdvanced();
+
+        // because the vault can call updateInvested => withdrawToVault multiple times during the same epoch,
+        // and rysk liquidity pool aggregates withdrawal amounts inititated during the same epoch,
+        // we cannot withdraw less than already pending amount
+        if (pendingWithdrawal.shares >= shares) return;
+
+        uint256 sharesToWithdraw = shares - pendingWithdrawal.shares;
+        if (!_hasEnoughShares(sharesToWithdraw))
+            revert StrategyNotEnoughShares();
+
+        pendingWithdrawal.shares += uint128(sharesToWithdraw);
+
+        emit RyskWithdrawalInitiated(sharesToWithdraw);
+        ryskLqPool.initiateWithdraw(sharesToWithdraw);
+    }
+
+    /**
+     * Completes the withdrawal initiated by withdrawToVault.
+     *
+     * @notice Expected to be called by the backend once the Rysk liquidity pool enters a new epoch.
+     * Backend should subscribe to 'EpochExecuted' event on Rysk LiquidityPool contract,
+     * and act by calling this function when the event is emitted to complete ongoing withdrawal.
+     */
+    function completeWithdrawal() external {
+        if (pendingWithdrawal.epoch == 0) revert RyskNoWithdrawalInitiated();
+        if (pendingWithdrawal.epoch == ryskLqPool.epoch())
+            revert RyskCannotComipleteWithdrawalInSameEpoch();
+
+        uint256 sharesToWithdraw = ryskLqPool
+            .withdrawalReceipts(address(this))
+            .shares;
+
+        uint256 amountWithdrawn = ryskLqPool.completeWithdraw(sharesToWithdraw);
+
+        emit StrategyWithdrawn(amountWithdrawn);
+        underlying.safeTransfer(vault, amountWithdrawn);
+    }
+
+    /**
+     * Get the number of Rysk liquidity pool shares owned by the strategy.
+     *
+     * @return shares owned by the strategy
+     */
+    function _getTotalShares() internal view returns (uint256) {
+        // total shares = redeemed shares + unredeemed shares
+        return
+            ryskLqPool.balanceOf(address(this)) +
```

Comment:
wondering if u would want to count pending withdrawals?

Reply:
I suppose not, since `initiateWithdraw` would transfer those shares to LiquidityPool and they will not generate any yield for us from that point. My thinking is that we should treat them as if they are already withdrawn and not count them as invested assets

Reply:
On second thought, I think you are right, we should include pending withdrawals, or else we will have temporary inconsistencies in estimating how much is invested by the strategy. I've already made the changes for counting pending withdrawal shares.